### PR TITLE
keyboard-layout-editor: init at 0-unstable-2019-05-14

### LINKED ISSUE #124134: Package the Keyboard Layout Editor software
**Project description**

From the package README:

> Keyboard layout editor is a JavaFX application made to help create custom keyboard layouts. The application works with the X Keyboard Extension (XKB).

**Metadata**

* source URL: [vgresak/keyboard-layout-editor](https://github.com/vgresak/keyboard-layout-editor)
* license: MIT
* platform: Linux
* build requirements: `jdk`, `openfjx` (already packaged)

### PR DESCRIPTION
**Pull request by FliegendeWurst**
Fixes #124134

Tested program, works well.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
@ofborg build keyboard-layout-editor

File: pkgs/by-name/ke/keyboard-layout-editor/package.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Keyboard layout editor for XKB";
+    homepage = "https://github.com/vgresak/keyboard-layout-editor";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ fliegendewurst ];
+    mainProgram = "keyboard-layout-editor";
+    platforms = jdk.meta.platforms;
```
Comment (misuzu) [inline pkgs/by-name/ke/keyboard-layout-editor/package.nix:69]: Looks broken on aarch64-linux
```
error: builder for '/nix/store/plk833gv2sv7ysyyla450jrmgfqd0ary-keyboard-layout-editor-0-unstable-2019-05-14.drv' failed with exit code 1;
       last 25 log lines:
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-fxml/17/javafx-fxml-17-linux-aarch64.jar
       >    > Could not find javafx-controls-17-linux-aarch64.jar (org.openjfx:javafx-controls:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-controls/17/javafx-controls-17-linux-aarch64.jar
       >    > Could not find javafx-graphics-17-linux-aarch64.jar (org.openjfx:javafx-graphics:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-graphics/17/javafx-graphics-17-linux-aarch64.jar
       >    > Could not find javafx-base-17-linux-aarch64.jar (org.openjfx:javafx-base:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/17/javafx-base-17-linux-aarch64.jar
       >
       > * Try:
       > > Run with --stacktrace option to get the stack trace.
       > > Run with --info or --debug option to get more log output.
       > > Run with --scan to get full insights.
       > > Get more help at https://help.gradle.org.
       >
       > Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
       >
       > You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
       >
       > For more on this, please refer to https://docs.gradle.org/8.10.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
       >
       > BUILD FAILED in 10s
       For full logs, run 'nix log /nix/store/plk833gv2sv7ysyyla450jrmgfqd0ary-keyboard-layout-editor-0-unstable-2019-05-14.drv'.
```

Comment (FliegendeWurst) [reply pkgs/by-name/ke/keyboard-layout-editor/package.nix:69]: Yeah, I asked about this on the JVM Matrix. The problem is that the resolved dependencies depend on the platform. So I'd have to combine all the deps.json. Another suggestion was to somehow use nixpkgs' openjfx package.

Comment (misuzu) [reply pkgs/by-name/ke/keyboard-layout-editor/package.nix:69]: Adding `broken = stdenv.hostPlatform.isAarch64;` might be good enough...

**Review by misuzu — APPROVED** [review]

**Pull request merged**